### DISCUSSION TITLE
How to check in HA is Onedrive backup is still running

### DISCUSSION THREAD

**Discussion by msd3287** [lavinir/hassio-onedrive-backup#235] [Ideas]
Hi,
Thanks for this nice addon.

Regularly, on my own HA instance and on my daughter's one, the Onedrive backup addon stop to work. Till now, I can't define why, and I must restart it manually.
Is there a way to check the running sate of the addon, and the command to start it within an automation? With this I would be able to create an automation to restart the addon after it stop... for the time I search the real cause of this unwanted stops.
Thanks

**Comment by lavinir** [comment]
Hi, the addon shouldn't just stop. You could follow the sensors it exposes to detect that backups are stale / not running but that shouldn't be the solution.
There is an issue reporting the same thing about the addon stopping here: #230  
Please track it there, or add anything that might be helpful to that thread.

**Comment by nickt444** [comment]
I had the same issue. I am using a Mushroom template badge on my homepage to show the age of last cloud backup and colour code accordingly. If I see > 1 day old, I usually need to restart the addon

for the badge colour:

{% set lastOnedriveBackup = state_attr('sensor.onedrivebackup', 'Last OneDrive backup date') | as_datetime | as_local %}
{% if lastOnedriveBackup > ( now() - timedelta(days=1) ) %}
  green
{% else %}
  red
{% endif %}

and for the badge content:

{% set lastOnedriveBackup = state_attr('sensor.onedrivebackup', 'Last OneDrive backup date') | as_datetime | as_local %}
{% set timeSinceLastBackup = time_since(lastOnedriveBackup, precision=1) %}
{{ timeSinceLastBackup }}
